recorderproxy/https.go

```diff
@@ -0,0 +1,295 @@
+			req, err := http.ReadRequest(clientTlsReader)
+			ctx := NewRecordContext(proxy)
+			ctx.init(req)
+			if err != nil && err != io.EOF {
+				return
+			}
+			if err != nil {
+				ctx.Warnf("Cannot read TLS request from mitm'd client %v %v", r.Host, err)
+				return
+			}
+			req.RemoteAddr = r.RemoteAddr // since we're converting the request, need to carry over the original connecting IP as well
+			ctx.Logf("req %v", r.Host)
+
+			if !httpsRegexp.MatchString(req.URL.String()) {
+				req.URL, err = url.Parse("https://" + r.Host + req.URL.String())
+			}
+
+			ctx.Req = req
+
+			req, resp := proxy.filterRequest(req, ctx)
+			if resp == nil {
+				if err != nil {
+					ctx.Warnf("Illegal URL %s", "https://"+r.Host+req.URL.Path)
+					return
+				}
+				removeProxyHeaders(ctx, req)
+				resp, err = ctx.proxy.RoundTripper.RoundTrip(req, ctx)
+				if err != nil {
+					ctx.Warnf("Cannot read TLS response from mitm'd server %v", err)
+					if err.Error() == "Connect failed" {
+						proxyClient.Close()
+					}
+					return
+				}
+				ctx.Logf("resp %v", resp.Status)
+			}
+			resp = proxy.filterResponse(resp, ctx)
+			defer resp.Body.Close()
+
+			text := resp.Status
+			statusCode := strconv.Itoa(resp.StatusCode) + " "
+			if strings.HasPrefix(text, statusCode) {
+				text = text[len(statusCode):]
+			}
+			// always use 1.1 to support chunked encoding
+			if _, err := io.WriteString(rawClientTls, "HTTP/1.1"+" "+statusCode+text+"\r\n"); err != nil {
+				ctx.Warnf("Cannot write TLS response HTTP status from mitm'd client: %v", err)
+				return
+			}
+			// Since we don't know the length of resp, return chunked encoded response
+			// TODO: use a more reasonable scheme
+			resp.Header.Del("Content-Length")
+			resp.Header.Set("Transfer-Encoding", "chunked")
+			// Force connection close otherwise chrome will keep CONNECT tunnel open forever
+			resp.Header.Set("Connection", "close")
+			if err := resp.Header.Write(rawClientTls); err != nil {
+				ctx.Warnf("Cannot write TLS response header from mitm'd client: %v", err)
+				ctx.SendErrorCode(-5011, "CANCELED_BY_BROWSER", "Veidemann recorder proxy lost connection to client")
+
+				return
+			}
+			if _, err = io.WriteString(rawClientTls, "\r\n"); err != nil {
+				ctx.Warnf("Cannot write TLS response header end from mitm'd client: %v", err)
+				return
+			}
+			chunked := newChunkedWriter(rawClientTls)
+			if _, err := io.Copy(chunked, resp.Body); err != nil {
+				ctx.Warnf("Cannot write TLS response body from mitm'd client: %v", err)
+				return
+			}
+			if err := chunked.Close(); err != nil {
+				ctx.Warnf("Cannot write TLS chunked EOF from mitm'd client: %v", err)
+				return
+			}
+			if _, err = io.WriteString(rawClientTls, "\r\n"); err != nil {
+				ctx.Warnf("Cannot write TLS response chunked trailer from mitm'd client: %v", err)
+				return
+			}
+		}
+		ctx.Logf("Exiting on EOF")
+	}()
+}
+
+func httpError(w io.WriteCloser, ctx *recordContext, err error) {
```

Comment:
`err` argument is unused in function body. Isn't the point of the function to handle the error in some way?